ssl/t1_lib.c

```diff
@@ -1196,13 +1522,35 @@ int ssl_setup_sig_algs(SSL_CTX *ctx)
             cache[i].enabled = 0;
         EVP_PKEY_CTX_free(pctx);
     }
+
+    /* Now complete cache and tls12_sigalgs list with provider sig information */
+    cache_idx = OSSL_NELEM(sigalg_lookup_tbl);
+    for (i = 0; i < ctx->sigalg_list_len; i++) {
+	TLS_SIGALG_INFO si = ctx->sigalg_list[i];
+	cache[cache_idx].name = si.name;
+	cache[cache_idx].sigalg = si.code_point;
+	tls12_sigalgs_list[cache_idx] = si.code_point;
+	cache[cache_idx].hash = si.hash_name?OBJ_sn2nid(si.hash_name):NID_undef;
+	cache[cache_idx].hash_idx = ssl_get_md_idx(cache[cache_idx].hash);
+	cache[cache_idx].sig = OBJ_sn2nid(si.sigalg_name);
+	cache[cache_idx].sig_idx = i + SSL_PKEY_NUM;
+	cache[cache_idx].sigandhash = OBJ_sn2nid(si.sigalg_name);
+	cache[cache_idx].curve = NID_undef;
+        /* all provided sigalgs are enabled by load */
+        cache[cache_idx].enabled = 1;
+	cache_idx++;
```

Comment:
NIT: replace leading tab with 8 spaces

Reply:
so changed in https://github.com/openssl/openssl/pull/19312/commits/1f20a63fbcb8f733651546e40328ae2ab2090537